holmes/plugins/toolsets/mcp/toolset_mcp.py

```diff
@@ -135,3 +216,75 @@ async def _get_server_tools(self):
 
     def get_example_config(self) -> Dict[str, Any]:
         return {}
+
+
+class StdioMCPToolset(BaseMCPToolset):
+    tools: List[StdioMCPTool] = Field(default_factory=list)  # type: ignore
+
+    def _create_tools(self, tools: List[MCP_Tool]) -> List[Tool]:
+        return [StdioMCPTool.create(self.stdio_server_params, tool) for tool in tools]
+
+    @property
+    def stdio_server_params(self) -> StdioServerParameters:
+        if not self.config:
+            raise ValueError("Config is required for stdio MCP server")
+        params = dict(self.config)
+        # pop out type which is not a parameter of StdioServerParameters
+        params.pop("type", None)
+        return StdioServerParameters(**params)
+
+    def _get_server_type(self) -> str:
+        return "stdio"
+
+    @property
+    def command(self) -> str:
+        return self.stdio_server_params.command
+
+    @property
+    def args(self) -> list[str]:
+        return self.stdio_server_params.args
+
+    def _get_connection_info(self) -> str:
+        return f"{self.command} {' '.join(self.args)}"
+
+    async def _get_server_tools(self):
+        server_params = StdioServerParameters(command=self.command, args=self.args)
+        async with stdio_client(server_params) as (
+            read_stream,
+            write_stream,
+        ):
+            async with ClientSession(read_stream, write_stream) as session:
+                _ = await session.initialize()
+                return await session.list_tools()
+
+    def get_example_config(self) -> Dict[str, Any]:
+        return {"command": "python", "args": ["/path/to/mcp_server.py"]}
+
+
+def get_mcp_toolset_from_config(config: dict[str, Any], name: str) -> BaseMCPToolset:
+    if not config:
+        raise ValueError("Config must not be empty")
+    # Prefer explicit mcp type set in nested 'config' for clarity.
+    # Supported values: 'sse' (remote SSE server) and 'stdio' (stdio server).
+    mcp_config = config.get("config", {})
+    mcp_type = mcp_config.get("type")
```

Comment:
Shouldn't the default be sse?

Reply:
If the user specify the url, it's ok but if the user set no type and no url, we'll raise ValidationError instead of 
`ValueError(
        "MCP Server config must include a transport type ('sse' or 'stdio') "
        "either in 'config.type' or by providing transport-specific keys."
    )`